src/main/java/net/snowflake/ingest/streaming/internal/ParquetFlusher.java

```
@@ -0,0 +1,271 @@
+        firstChannel = data.getChannel();
+      } else {
+        // This method assumes that channelsDataPerTable is grouped by table. We double check
+        // here and throw an error if the assumption is violated
+        if (!data.getChannel()
+            .getFullyQualifiedTableName()
+            .equals(firstChannel.getFullyQualifiedTableName())) {
+          throw new SFException(ErrorCode.INVALID_DATA_IN_CHUNK);
+        }
+
+        columnEpStatsMapCombined =
+            ChannelData.getCombinedColumnStatsMap(columnEpStatsMapCombined, data.getColumnEps());
+      }
+      rows.addAll(data.getVectors().rows);
+
+      rowCount += data.getRowCount();
+
+      logger.logDebug(
+          "Parquet Flusher: Finish building channel={}, rowCount={}, bufferSize={} in blob={}",
+          data.getChannel().getFullyQualifiedName(),
+          data.getRowCount(),
+          data.getBufferSize(),
+          filePath);
+    }
+
+    Map<String, String> metadata = channelsDataPerTable.get(0).getVectors().metadata;
+
+    flushToParquet(chunkData, rows, metadata);
+    return new SerializationResult(channelsMetadataList, columnEpStatsMapCombined, rowCount);
+  }
+
+  private void flushToParquet(
+      ByteArrayOutputStream output, List<List<Object>> rows, Map<String, String> metadata)
+      throws IOException {
+    try {
+      ParquetWriter<List<Object>> writer =
+          new BdecParquetWriterBuilder(output, schema, metadata)
+              .withWriterVersion(ParquetProperties.WriterVersion.PARQUET_1_0)
+              .withDictionaryEncoding(false)
+              .enableValidation()
+              .withCompressionCodec(CompressionCodecName.UNCOMPRESSED)
+              .withWriteMode(ParquetFileWriter.Mode.CREATE)
+              .build();
+
+      for (List<Object> row : rows) {
+        writer.write(row);
+      }
+      writer.close();
+    } catch (Throwable t) {
+      logger.logError("Parquet Flusher: failed to write", t);
+      throw t;
```

Why do we need to catch it and log it? Can't we just let it throw?

---

I think I wanted to explicitly log it in case of internal errors.

---

Is there something above this that catches/logs it?

---

My problem was that I did not see the error cause in the logs, maybe this has changed